#!/bin/sh /usr/share/dpatch/dpatch-run
## 13-module-mismatch.dpatch by Daniel Baumann <daniel@debian.org>
##
## DP: Adjusts failure message with Debian specific solution.

@DPATCH@

diff -Naurp virtualbox-ose-1.6.2-dfsg.orig/src/VBox/VMM/VM.cpp virtualbox-ose-1.6.2-dfsg/src/VBox/VMM/VM.cpp
--- virtualbox-ose-1.6.2-dfsg.orig/src/VBox/VMM/VM.cpp	2008-05-14 15:09:39.000000000 +0000
+++ virtualbox-ose-1.6.2-dfsg/src/VBox/VMM/VM.cpp	2008-06-23 16:10:30.000000000 +0000
@@ -293,9 +293,11 @@ VMR3DECL(int)   VMR3Create(PFNVMATERROR 
                     break;
                 case VERR_VERSION_MISMATCH:
                 case VERR_VM_DRIVER_VERSION_MISMATCH:
-                    pszError = N_("The VirtualBox support driver which is running is from a different "
-                                  "version of VirtualBox.  You can correct this by stopping all "
-                                  "running instances of VirtualBox and reinstalling the software.");
+                    pszError = N_("The version of the VirtualBox kernel modules and the version of "
+				  "VirtualBox application are not matching. You can correct this by "
+				  "either installing the correct virtualbox-ose-modules package "
+				  "(if available) through apt-get or by building it manually with: "
+				  "module-assistant auto-install virtualbox-ose");
                     break;
                 default:
                     pszError = N_("Unknown error initializing kernel driver");
